rust/src/checkpoints.rs

```diff
@@ -100,6 +122,36 @@ pub async fn create_checkpoint_from_table(table: &DeltaTable) -> Result<(), Chec
         &table.table_uri,
     )
     .await?;
+
+    let new_version = table.version + 1;
+
+    if new_version > 0 {
+        let deleted_log_num = cleanup_expired_logs(
+            new_version,
```

Comment:
do we need that variable? at first look it confuses, e.g. since you're not creating `new` version

Reply:
yea, the variable can be removed.